sdk/core/Azure.Core/src/DynamicData/DynamicData.AllowList.cs

```diff
@@ -0,0 +1,202 @@
+            }
+
+            private static bool IsAllowedPrimitive(Type type)
+            {
+                return
+                    type.IsPrimitive ||
+                    type == typeof(decimal) ||
+                    type == typeof(string) ||
+                    type == typeof(DateTime) ||
+                    type == typeof(DateTimeOffset) ||
+                    type == typeof(TimeSpan) ||
+                    type == typeof(Uri) ||
+                    type == typeof(Guid) ||
+                    type == typeof(ETag);
+            }
+
+            private static bool IsAllowedArrayType(Type type)
+            {
+                if (!type.IsArray)
+                {
+                    return false;
+                }
+
+                Type? elementType = type.GetElementType();
+                return elementType != null && IsAllowedType(elementType);
+            }
+
+            private static bool IsAllowedCollectionType(Type type)
+            {
+                return
+                    IsAllowedListType(type) ||
+                    IsAllowedDictionaryType(type);
+            }
+
+            private static bool IsAllowedListType(Type type)
+            {
+                if (!type.IsGenericType)
+                {
+                    return false;
+                }
+
+                if (type.GetGenericTypeDefinition() != typeof(List<>))
+                {
+                    return false;
+                }
+
+                Type[] types = type.GetGenericArguments();
+                return IsAllowedType(types[0]);
+            }
+
+            private static bool IsAllowedDictionaryType(Type type)
+            {
+                if (!type.IsGenericType)
+                {
+                    return false;
+                }
+
+                if (type.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+                {
+                    Type[] types = type.GetGenericArguments();
+                    if (IsAllowedType(types[0]) && IsAllowedType(types[1]))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            private static bool IsAllowedEnumerableType(Type type)
+            {
+                if (!type.IsGenericType)
+                {
+                    return false;
+                }
+
+                if (!type.IsInterface)
+                {
+                    return false;
+                }
+
+                if (type.GetGenericTypeDefinition() != typeof(IEnumerable<>))
+                {
+                    return false;
+                }
+
+                Type[] types = type.GetGenericArguments();
+                return IsAllowedType(types[0]);
+            }
+
+            private static bool IsAllowedPocoType(Type type, HashSet<Type> ancestorTypes)
+            {
+                if (!HasPublicParameterlessConstructor(type) && !IsAnonymousType(type))
```

Comment:
My goal here was to create a filter that could differentiate between allowable and non-allowable types.  I used having a public constructor and read-write public properties as a way to identify serializable POCOs, but since anonymous types don't have public constructors or writable public properties, I let them through differently.  I'm entirely open to other approaches here if you want to suggest one that separates types as we intend.

Reply:
Since we are running all these checks to see whether the types are serializable, I don't think we need setters or the ctor.

Reply:
Cool, discussed offline.  I'll see if I can find a better filter algorithm for this.

Reply:
Removed support for POCO assignment for v1.